include/z64save.h

```diff
@@ -4,6 +4,32 @@
 #include "ultra64.h"
 #include "z64math.h"
 
+typedef enum MagicBarAction {
+    /* 00 */ MAGIC_BAR_ACTION_IDLE, // Regular gameplay
+    /* 01 */ MAGIC_BAR_ACTION_CONSUME_SETUP, // Sets the speed in which magic border flashes
+    /* 02 */ MAGIC_BAR_ACTION_CONSUME, // Consume magic until target is reached or no more magic is available
+    /* 03 */ MAGIC_BAR_ACTION_BORDER_CHANGE_1, // Flashes border and freezes Dark Link
+    /* 04 */ MAGIC_BAR_ACTION_BORDER_CHANGE_2, // Flashes border and draws yellow magic to preview target consumption
+    /* 05 */ MAGIC_BAR_ACTION_RESTORE_IDLE, // Reset colors and return to idle
+    /* 06 */ MAGIC_BAR_ACTION_BORDER_CHANGE_3, // Flashes border with no additional behaviour
+    /* 07 */ MAGIC_BAR_ACTION_LENS_CONSUME, // Magic slowly consumed by lens. 
+    /* 08 */ MAGIC_BAR_ACTION_GROW_WIDE, // Init magic on a new load, grow from a width of 0 to magicCapacity
+    /* 09 */ MAGIC_BAR_ACTION_FILL, // Add magic until full capacity is reached
+    /* 10 */ MAGIC_BAR_ACTION_ADD // Add requested magic
+} MagicBarAction;
```

Comment:
I first named it "state", so I'm not opposed to the name. I decided to switch to "action" because it seemed like it was describing the magic bar actually "doing" something as oppose to "state" which sounds more like a passive description.

I would guess "state" is probably more consistent with other things in the repo, although you'd have a better sense of that

Reply:
I guess let's take a look at the spin attack as an example.

This is the line where you specify you want magic to be consumed. But this only sets the target, and previews that target with the yellow bar drawn between `magic` and `magicTarget` and it remains that way until an external trigger occurs
https://github.com/zeldaret/oot/blob/764a102a863d828c3d72227a49bac6d553b60bd9/src/overlays/actors/ovl_En_M_Thunder/z_en_m_thunder.c#L162

And that trigger is this line below. Once the magic is ready to actually be consumed, the action is set, triggering the magic to be consumed. It's almost like this variable causes the magic bar to "act" in certain ways. It felt like it was being used more than just a state, it's actually used as a trigger to get the magic bar to act.
https://github.com/zeldaret/oot/blob/764a102a863d828c3d72227a49bac6d553b60bd9/src/overlays/actors/ovl_En_M_Thunder/z_en_m_thunder.c#L196

However, I will say that if state is used like this in other places and that you can set the state to trigger certain events to occur, instead of the state being used to passively describe what's happening to a system, then I might be fine with state

Reply:
action to me is just very specifically a word to describe what something more "physical" is doing, like an actor that is representing a real world thing. For a more abstract thing under the hood, state or mode seems more appropriate to me. 
I used mode a bunch in the file select system to describe the various things its doing. But tharo used state to describe the message state which i feel is pretty analogus here. providing examples from things by the same dev cause this guy like to assign state to numbers in this way pretty consistently lol